assignments/Assignment4.js

```diff
@@ -0,0 +1,48 @@
+
+// #2 Return the largest of 3 arguments numeric or string p.112
+function max3(x, y, z) {
+    if ((x > y) && (x > z)) {
+        return console.log(x + " is the largest");
+    } else if ((y > x) && (y > z)) {
+        return console.log(y + " is the largest");
+    } else {
+        return console.log(z + " is the largest");
+    }
+}
+
+max3(30, 50, 20)
+max3("bite", "bat", "bit")
+max3(3.14, 3.141, -3.145)
+
+
+
+// #5 Sum of odd integers p.112
+function sumOfOddIntegers(n) {
+    let sum = 0;
+    let ctr = 1;
+    // Start from 1 and increment by 2 fro odd numbers
+    for(let i = 1; ctr <= n; i += 2) {
```

Comment:
This is correct but it is very (very) unusual to not use the same variable in all three parts of the `for` loop - in this case that would be `i`.

Also, `i` is almost always used as a simple index or counter variable and any other state you need to maintain should be in a separate, appropriately named variable.

You can probably get away with having only two variables in fact, `i` for the simple counter and `sum` as the aggregator.  If I asked you to just tell me what the nth odd number is, can you find that based on a simple formula?

Reply:
I tried to do that but  I couldn't get it to work correctly until I broke out a separate counter that tracked the iterations of the number initially entered (N = 4) and the i ctr kept track of the odd numbers needed to be added together. For example, N= 4 equals 1 + 3+ 5+7 = 16. When I enter 5 = result is 25. 
function sumOfOddIntegers(n) {
    let sum = 0;
    let ctr = 1;
    // Start from 1 and increment by 2 fro odd numbers
    for(let i = 1; ctr <= n; i += 2) {
        sum = sum + i;
        ctr = ctr + 1;
    }
    return sum;
  
  }

  console.log("The sum of odd numbers for this integer is: " + sumOfOddIntegers(4));
  console.log("The sum of odd numbers for this integer is: " + sumOfOddIntegers(5));
  console.log("The sum of odd numbers for this integer is: " + sumOfOddIntegers(6));

RESULT:
The sum of odd numbers for this integer is: 16
The sum of odd numbers for this integer is: 25
The sum of odd numbers for this integer is: 36